Airship/AirshipCore/Source/DefaultLogHandler.swift

```diff
@@ -0,0 +1,44 @@
+/* Copyright Airship and Contributors */
+
+import Foundation
+import os
+
+/**
+ * Default log handler. Logs to either os.Logger or just prints depending on OS version.
+ */
+class DefaultLogHandler: AirshipLogHandler {
+    @available(macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
+    static var logger: Logger = Logger(subsystem: Bundle.main.bundleIdentifier ?? "", category: "Airship")
+
+    func log(logLevel: LogLevel, message: String, fileID: String, line: UInt, function: String) {
+        if #available(macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0, *) {
+            DefaultLogHandler.logger.log(level: logLevel.logType, "[\(logLevel.initial)] \(fileID) \(function) [Line \(line)] \(message)")
+        } else {
+            print("[\(logLevel.initial)] \(fileID) \(function) [Line \(line)] \(message)")
+        }
+    }
+}
+
+private extension LogLevel {
+    var initial: String {
+        switch self {
+        case .trace: return "T"
+        case .debug: return "D"
+        case .info: return "I"
+        case .warn: return "W"
+        case .error: return "E"
+        default: return "U"
+        }
+    }
+
+    var logType: OSLogType {
+        switch self {
+        case .trace: return OSLogType.debug
+        case .debug: return OSLogType.debug
+        case .info: return OSLogType.info
+        case .warn: return OSLogType.info
```

Comment:
I might suggest changing this one to `default`.
"Use this level to capture information about things that might result in a failure."

`info` won't necessarily get saved long term if an issue happens due to internal buffering strategy.
"Use this level to capture information that may be helpful, but not essential, for troubleshooting errors."

I know this approach matches this, but I also disagree with this choice. `Warn` is higher severity than `Notice` in SwiftLog terms: https://github.com/chrisaljoudi/swift-log-oslog/blob/master/Sources/LoggingOSLog/LoggingOSLog.swift#L83

I understand this is the default implementation which I will override in my usage, but just something to consider. :)

Reply:
👍